devices/libsdio/README.md

```diff
@@ -0,0 +1,277 @@
+
+Following sections describe the API in detail, including the behaviour of certain functions.
+
+---
+
+```c
+typedef enum {
+	sdio_read,
+	sdio_write
+} sdio_dir_t;
+```
+
+This enumeration type is meant as parameter for `sdio_transferDirect` and `sdio_transferBulk` functions to indicate the direction of a given data transfer.
+
+---
+
+```c
+typedef void (*sdio_event_handler_t)(void *arg);
+```
+
+This type describes a function pointer for an interrupt handler. Handlers of interrupts events are assigned to a specific event, and every event can only have a single handler. Handler argument is provided during handler registration using `sdio_eventRegister`, and will remain unchanged until the handler registration is modified via the same function.
+
+---
+
+```c
+int sdio_init(void);
+```
+
+This function initializes the SDIO inteface hardware and tries to detect and enable connected device. Should no device be present or it could not respond to SDIO initialization sequence, this function will return an error code and free unusued resources it acquired.
+
+> **NOTE:** This function should not be called multiple times. In order to reinitialize the interface, `sdio_free` should be called between concurrent `sdio_init` calls instead.
+
+Parameters:
+
+- none
+
+Returns:
+
+| code | description |
+|------|-------------|
+|  0   | success     |
+| -1   | failure     |
+
+> **NOTE:** Every failure is accompanied by explicit fault description printed into standard output.
+
+---
+
+```c
+void sdio_free(void);
+```
+
+This function will free the SDIO bus, deregister and disable all previously registered event handlers.
+
+> **NOTE:** Due to its current implementation, calling this function does not allow for creation of new library instances in other running processes.
+
+Parameters:
+
+- none
+
+Returns:
+
+- nothing
+
+---
+
+```c
+int sdio_config(uint32_t freq, uint16_t blocksz);
+```
+
+This function provides a configuration interface for the SDIO bus.
+
+Parameters:
+
+| parameter               | description                   |
+|-------------------------|-------------------------------|
+| [in] `uint32_t freq`    | SDIO clock frequency in Hz    |
+| [in] `uint16_t blocksz` | block size for bulk transfers |
+
+> **NOTE:** `uint32_t freq` can only be specified as either 25000000 or 50000000, since only 25 and 50 MHz clock speeds are supported.
+
+Returns:
+
+| code | description |
+|------|-------------|
+|  0   | success     |
+| -1   | failure     |
+
+> **NOTE:** Every failure is accompanied by explicit fault description printed into standard output.
+
+---
+
+```c
+int sdio_transferDirect(sdio_dir_t dir, uint32_t address, uint8_t area, uint8_t *data);
+```
+
+This function initiates a direct, single 8-bit register read/write opetation. When `dir` is specified as `sdio_write` the byte inside the `data` buffer is written to the card, otherwise if `dir` equals `sdio_read` is specified, the buffer is set to value read from the card.
```

Comment:
```suggestion
This function initiates a direct, single 8-bit register read/write opetation. When `dir` is specified as `sdio_write` the byte inside the `data` buffer is written to the device, otherwise if `dir` equals to `sdio_read` the buffer is set to value read from the device.
```

Reply:
Changed as requested.